pkg/apis/workflow/v1alpha1/container_set_template_types.go

```diff
@@ -14,6 +14,7 @@ type ContainerSetTemplate struct {
 	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty" protobuf:"bytes,3,rep,name=volumeMounts"`
 	// RetryStrategy describes how to retry a container nodes in the container set if it fails.
 	// Nbr of retries(default 0) and sleep duration between retries(default 0s, instant retry) can be set.
+	// The container won't retry if it's unable to locate the command.
 	RetryStrategy *ContainerSetRetryStrategy `json:"retryStrategy,omitempty" protobuf:"bytes,5,opt,name=retryStrategy"`
 }
 
```

Comment:
```suggestion

// ContainerSetRetryStrategy provides controls on how to retry a container set
```

A comment should be added above this struct too as it will appear in the Fields reference as well -- it's [currently empty](https://argo-workflows.readthedocs.io/en/release-3.5/fields/#containersetretrystrategy):
![Screenshot 2024-03-26 at 3 20 33 AM](https://github.com/argoproj/argo-workflows/assets/4970083/76298f9f-d6d4-4a98-82aa-b15919f1ba41)

This comment should similarly match the other template types' `retryStrategy` struct [description](https://github.com/argoproj/argo-workflows/blob/24f791b9f5d9c3d946670366e1c86d802ea48ce1/pkg/apis/workflow/v1alpha1/workflow_types.go#L2030C4-L2030C67)

Reply:
Should also modify line 25 `// Nbr of retries` to match the [description of `limit`](https://github.com/argoproj/argo-workflows/blob/24f791b9f5d9c3d946670366e1c86d802ea48ce1/pkg/apis/workflow/v1alpha1/workflow_types.go#L2034).

Honestly it should also be renamed, but that's a whole separate discussion -- the entire spec should ideally be identical to other template types instead of being its own separate thing.

Reply:
Follow up [comment](https://github.com/argoproj/argo-workflows/pull/12809#discussion_r1539774685) for `retries` description